src/Raven.Studio/typescript/models/database/tasks/ongoingTaskSubscriptionListModel.ts

```diff
@@ -26,19 +33,60 @@ class ongoingTaskSubscriptionListModel extends ongoingTaskListModel {
 
     validationGroup: KnockoutValidationGroup; 
     showDetails = ko.observable(false);
+   
+    documentIDForNextBatchStartingPoint = ko.observable<string>(null);
+    nextBatchStartingPointDocumentFound = ko.observable<boolean>(false);
+    triedToFindDocumentNextBatchButFailed = ko.observable<boolean>(false);
+    urlForNextBatchStartingPointDocument: KnockoutComputed<string>;
+
+    documentIDForLastChangeVectorAcknowledged = ko.observable<string>(null);
+    lastAcknowledgedDocumentFound  = ko.observable<boolean>(false);
+    triedToFindDocumentLastAcknowledgedButFailed = ko.observable<boolean>(false);
+    urlForLastAcknowledgedDocument: KnockoutComputed<string>;
+
+    spinners = {
+        documentIDForNextBatchStartingPointLoading: ko.observable<boolean>(false),
+        documentIDForLastChangeVectorAcknowledgedLoading: ko.observable<boolean>(false)
+    };
     
     constructor(dto: Raven.Client.Documents.Operations.OngoingTasks.OngoingTaskSubscription) {
         super();
 
         this.update(dto);
-        this.initializeObservables(); 
+        this.initializeObservables();
+        this.bindToCurrentInstance("tryGetDocumentIDFromChangeVector");
+    }
+
+    protected bindToCurrentInstance(...methods: Array<keyof this>) {
+        _.bindAll(this, ...methods);
     }
 
     initializeObservables() {
         super.initializeObservables();
 
+        this.spinners.documentIDForNextBatchStartingPointLoading.extend({ rateLimit: 200});
+        this.spinners.documentIDForLastChangeVectorAcknowledgedLoading.extend({ rateLimit: 200});
+
         const urls = appUrl.forCurrentDatabase();
         this.editUrl = urls.editSubscription(this.taskId, this.taskName());
+
+        this.changeVectorForNextBatchStartingPointFormatted = ko.pureComputed(() => {
```

Comment:
code duplication

Reply:
done - created a method that returns a pure computed